src/js/util/svg.js

```diff
@@ -54,40 +54,33 @@ define(function (require, exports) {
      *
      * @private
      * @param {Array.<string>} filter
-     * @return {Array.<string>}
+     * @return {string}
     */
     var getSVGClassesFromFilter = function (filter) {
         if (filter.length === 0) {
-            return [];
+            return null;
         }
         // currently only have icons for layers
         if (filter.length < 2 || filter.join(" ").indexOf("LAYER") === -1) {
-            return ["tool-rectangle"]; // standin for non-layers
+            return "tool-rectangle"; // standin for non-layers
         }
 
-        var iconIDs = [],
+        var iconID = "layer-",
             isLinked = _.has(filter, "linked");
-
+        
         _.forEach(filter, function (kind) {
-            var iconID = "layer-";
-
             if (kind === "ARTBOARD") {
                 iconID += "artboard";
             } else if (kind === "BACKGROUND") {
                 iconID += layerLib.layerKinds.PIXEL;
             } else if (kind === "SMARTOBJECT" && isLinked) {
                 iconID += layerLib.layerKinds.SMARTOBJECT + "-linked";
-            } else if (kind !== "LAYER") {
+            } else if (kind.indexOf("LAYER") === -1) { // check if kind is "ALL_LAYER" or "CURRENT_LAYER"
                 iconID += layerLib.layerKinds[kind.toUpperCase().replace(" ", "")];
             }
```

Comment:
That shouldn't ever happen. If the filter just contains "ALL_LAYER" or "CURRENT_LAYER" then it returns above. And otherwise the options are all in layerKinds, so they'll give the correct value.

I realize the logic to get the icons is very layer-specific right now, but once I get the other SVGs from Seth, I'll work on making it all more generic.